src/Controller/Admin/UserCrudController.php

```diff
@@ -82,4 +98,63 @@ public function configureCrud(Crud $crud): Crud
     {
         return $crud->setDefaultSort(['id' => 'DESC']);
     }
+
+    public function configureFilters(Filters $filters): Filters
+    {
+        //@todo update after merge https://github.com/EasyCorp/EasyAdminBundle/pull/3725/files
+//        $rolesFilter = ArrayFilter::new('roles')
+//            ->setChoices(self::ROLES_AVAILABLE)
+//            ->canSelectMultiple(true)
+//        ;
+
+        $rolesFilter = ChoiceFilter::new('roles')
+            ->setChoices(self::ROLES_AVAILABLE)
+            ->canSelectMultiple(true)
+        ;
+
+        $rolesFilter
+            ->getAsDto()
+            ->setApplyCallable(static function (QueryBuilder $queryBuilder, FilterDataDto $filterDataDto, ?FieldDto $fieldDto, EntityDto $entityDto) {
+                $alias = $filterDataDto->getEntityAlias();
+                $property = $filterDataDto->getProperty();
+                $comparison = $filterDataDto->getComparison();
+                $parameterName = $filterDataDto->getParameterName();
+                $value = $filterDataDto->getValue();
+                $isMultiple = $filterDataDto->getFormTypeOption('value_type_options.multiple');
+
+                if ('NOT IN' === $comparison) {
+                    $comparison = 'NOT LIKE';
+                } else {
+                    $comparison = 'LIKE';
+                }
+
+                if (null === $value || ($isMultiple && 0 === \count($value))) {
+                    $queryBuilder->andWhere(sprintf('%s.%s %s', $alias, $property, $comparison));
+                } else {
+                    $orX = new Orx();
+                    if ($isMultiple && is_array($value)) {
+                        foreach ($value as $i => $val) {
```

Comment:
Maybe it will be better to transfer it to another method

Reply:
I think i will remove this temporary code, after fixed issue from easyadmin